onnxruntime/core/optimizer/gemm_activation_fusion.cc

```diff
@@ -65,6 +65,12 @@ Status GemmActivationFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
       continue;
     }
 
+    NodeArg* node_output = node.MutableOutputDefs()[0];
```

Comment:
Would be good to add some background info in a comment here about what fp16 kernels/activations we have, why we can't fuse those currently, and if possible what would be required to enable fusion

Reply:
Search "DEFINE_ELE_KERNEL" will get all activations that can be supported.

Gelu is very popular but not supported in FusedGemm.

Reply:
There are already fp16 kernels for Relu and LeakyRelu that in theory could be fused with the fp16 Gemm, but there were issues when that was enabled. It's that info I'd like to capture in a comment as it's unexpected that we can't do those fusions already.

Reply:
Added comment and enabled fp16 fusedgemm for rely/leakyrelu

Reply:
Not sure whether it is good to fuse fp16 for Relu and LeakyRelu.  Do we have performance test shows that this actually help performance?

Tracy's description in https://github.com/microsoft/onnxruntime/pull/1403:
```
I started doing the same for FusedGemm, but I would rather see this fusion go away as it currently has zero benefit. None of the GEMMs currently supported a fused activation and it isn't a priority. 
```

Reply:
Oh i see. Disabled the fusion for now. Need this change with some urgency so we can readd additional data types if needed later.